src/combinator.rs

```diff
@@ -625,3 +642,41 @@ impl<I: Clone, O, A: Parser<I, O, Error = E>, U: Clone, E: Error<I>> Parser<I, U
     #[inline]
     fn parse_inner_silent(&self, d: &mut Silent, s: &mut StreamOf<I, E>) -> PResult<I, U, E> { #[allow(deprecated)] self.parse_inner(d, s) }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
I appreciate the tests. This combinator is by far the most complicated and I often have a hard time wrapping my head around the logic going on.